Tasks/PerformActionOnMachineGroup/PerformActionOnMachineGroup.ps1

```diff
@@ -54,10 +57,20 @@ Switch ($Action)
          break
       }
 
-     { @("Start", "Stop", "Restart") -contains $_ } {
+      { @("Start", "Stop", "Restart") -contains $_ } {
          Invoke-OperationHelper -machineGroupName $MachineGroupName -operationName $Action -machines $machineGroup.Resources
          break
       }
+ 
+      "Block" {
+          Block-MachineGroup -machineGroupName $MachineGroupName -blockedFor $BlockedFor -timeInHours $TimeInHours
+          break
+      }
+  
+      "Unblock" {
```

Comment:
This will also need to be factored in localized task changes. Please keep a note of it.